nncf/experimental/post_training/algorithms/quantization/fast_bias_correction.py

```diff
@@ -0,0 +1,225 @@
+from nncf.experimental.post_training.algorithms.algorithm import PostTrainingAlgorithms
+from nncf.experimental.post_training.api.engine import Engine
+from nncf.experimental.post_training.factories import NNCFGraphFactory
+from nncf.experimental.post_training.factories import PTQLayerMetatypesFactory
+from nncf.experimental.post_training.factories import PTQModelExtractionCommandFactory
+from nncf.experimental.post_training.factories import PTQNNCFTensorFactory
+from nncf.experimental.post_training.factories import PTQBiasCorrectionCommandFactory
+from nncf.experimental.post_training.factories import PTQTargetPointFactory
+from nncf.experimental.post_training.factories import PTQMeanStatisticCollectorFactory
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPoint
+from nncf.experimental.post_training.statistics.statistic_point import StatisticPointsContainer
+
+ModelType = TypeVar('ModelType')
+
+# pylint: disable=protected-access
+
+
+class FastBiasCorrectionParameters(AlgorithmParameters):
+    """
+    Base class of FastBiasCorrection parameters.
+    """
+
+    def __init__(self, number_samples: int = 100, threshold: float = 2.0):
+        self.number_samples = number_samples
+        self.threshold = threshold
+
+    def to_json(self) -> Dict[str, Union[str, float, int]]:
+        """
+        Serialize all FastBiasCorrection parameters to JSON.
+        """
+
+
+class FastBiasCorrection(Algorithm):
+
+    def __init__(self, parameters: FastBiasCorrectionParameters):
+        super().__init__()
+        self.number_samples = parameters.number_samples
+        self.threshold = parameters.threshold
+        self.nncf_graph = None
+        self._target_points_to_correct = []  # type: List[TargetPoint]
+        self._channel_axis_by_types = {
+            'Conv': 1,
+            'Gemm': -1,
+        }
+
+    def _generate_stat_collector(self, model_backend, axes) -> TensorStatisticCollectorBase:
+        return PTQMeanStatisticCollectorFactory.create(
+            model_backend,
+            reduction_shape=axes,
+            num_samples=self.number_samples)
+
+    @staticmethod
+    def _input_filter_func(point):
+        return PostTrainingAlgorithms.FastBiasCorrection in point.algorithm_to_tensor_collectors and \
+            point.target_point.type == TargetType.PRE_LAYER_OPERATION
+
+    @staticmethod
+    def _output_filter_func(point):
+        return PostTrainingAlgorithms.FastBiasCorrection in point.algorithm_to_tensor_collectors and \
+            point.target_point.type == TargetType.POST_LAYER_OPERATION
+
+    def _apply(self, model: ModelType, engine: Engine,
+               statistic_points: StatisticPointsContainer) -> ModelType:
+
+        model_backend = infer_backend_from_model(model)
+
+        transformation_layout = TransformationLayout()
+        nncf_graph = NNCFGraphFactory.create(model)
+
+        layers_with_bias_types = PTQLayerMetatypesFactory.get_layers_with_bias_types(
```

Comment:
Is this really a backend specific?

Reply:
To be true: I'm not sure. But as I can see, this was made similarly with the other backends. Here are the ONNX meta types - https://github.com/openvinotoolkit/nncf/blob/develop/nncf/experimental/onnx/graph/metatypes/onnx_metatypes.py
And here are the PT meta types, for example - https://github.com/openvinotoolkit/nncf/blob/develop/nncf/torch/graph/operator_metatypes.py